### PR TITLE
Error alert

### PR DESCRIPTION
**Pull request by arcelena**

### CODE REVIEW

File: mobile/src/reducers/organization/organization.epics.ts [hunk 1]
```diff
@@ -5,36 +5,49 @@ import { Department } from './department.model';
...
-        .map(x => 
+        .map(x =>
             x.switchMap(y =>
                 ajaxGetJSON(`${url}/employees?departmentId=${x.key}`).map(obj => deserializeArray(obj as any, Employee))))
         .mergeAll()
-        .flatMap(x => x.map(loadEmployeeFinished));
\ No newline at end of file
+        .flatMap(x => x.map(loadEmployeeFinished))
+        .catch(e => { 
+            console.log('error is catched');
```
Comment (santee) [inline mobile/src/reducers/organization/organization.epics.ts]: no need for console.log or braces

File: mobile/src/reducers/organization/organization.epics.ts [hunk 2]
```diff
@@ -5,36 +5,49 @@ import { Department } from './department.model';
...
     action$.ofType('LOAD-DEPARTMENTS-FINISHED')
-        .switchMap(x => 
+        .switchMap(x =>
             x.departments.map(dep =>
                 ajaxGetJSON(`${url}/employees/${dep.chiefId}`).map(obj => deserialize(obj, Employee))))
         .mergeAll()
-        .map(x => loadEmployeeFinished(x));
+        .map(x => loadEmployeeFinished(x))
+        .catch(e => { 
+            console.log('error is catched');
```
Comment (santee) [inline mobile/src/reducers/organization/organization.epics.ts]: no need for console.log

File: mobile/src/reducers/organization/organization.epics.ts [hunk 3]
```diff
@@ -5,36 +5,49 @@ import { Department } from './department.model';
...
 const url = 'http://localhost:5000/api'; //TODO: fix hardcode
 
 export const loadDepartmentsEpic$ = (action$: ActionsObservable<LoadDepartments>) =>
     action$.ofType('LOAD-DEPARTMENTS')
         .switchMap(x => ajaxGetJSON(`${url}/departments`))
         .map(x => deserializeArray(x as any, Department))
-        .map(x => loadDepartmentsFinished(x));
+        .map(x => loadDepartmentsFinished(x))
+        .catch(e => { 
+            console.log('error is catched');
```
Comment (santee) [inline mobile/src/reducers/organization/organization.epics.ts]: remove console.log, it's already logged by redux-logger

File: mobile/src/reducers/errorHandling/error.reducer.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { Reducer, combineReducers } from 'redux';
+import { OrganizationActions } from '../organization/organization.action';
+import { Alert } from 'react-native';
+
+var showAlert: boolean = false;
```
Comment (santee) [inline mobile/src/reducers/errorHandling/error.reducer.ts]: no var at any time. Also, run `yarn tslint`, i think there will be errors

File: mobile/src/reducers/errorHandling/error.reducer.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+function getAlert(message: string) {
+    Alert.alert('Error', message);
```
Comment (santee) [inline mobile/src/reducers/errorHandling/error.reducer.ts]: this code should not be here, it's supposed to be in tsx. I'll take a minute later, maybe we *can* put it in epics as you wanted at the beginning... but definitely not in reducers

File: mobile/src/reducers/organization/organization.epics.ts [hunk 6]
```diff
@@ -5,36 +5,50 @@ import { Department } from './department.model';
...
 const url = 'http://localhost:5000/api'; //TODO: fix hardcode
 
 export const loadDepartmentsEpic$ = (action$: ActionsObservable<LoadDepartments>) =>
     action$.ofType('LOAD-DEPARTMENTS')
         .switchMap(x => ajaxGetJSON(`${url}/departments`))
         .map(x => deserializeArray(x as any, Department))
-        .map(x => loadDepartmentsFinished(x));
+        .map(x => loadDepartmentsFinished(x))
+        .catch(e => { 
+            Alert.alert('Error', 'Server is not available')
```
Comment (santee) [inline mobile/src/reducers/organization/organization.epics.ts]: Ok, let's do the following

Comment (santee) [reply mobile/src/reducers/organization/organization.epics.ts]: remove Alert.alert from here

Comment (santee) [reply mobile/src/reducers/organization/organization.epics.ts]: add an Action interface and action creator for these errors

Comment (santee) [reply mobile/src/reducers/organization/organization.epics.ts]: Add a separate epic that listens all such error actions and makes Alert call inside its `map` function. We'll need to check though that it's really called and nothing breaks :)

Comment (santee) [reply mobile/src/reducers/organization/organization.epics.ts]: Can you try this code as said 'separate epic'? 
```
export const errorEpic= action$ =>
  action$.ofType(MY_MEGA_ERROR)
    .do(({ payload }) =>
      Alert.alert('Error', `error occurred: ${payload.message}`)
    )
    .ignoreElements();
```

File: mobile/src/reducers/organization/organization.epics.ts [hunk 7]
```diff
@@ -1,40 +1,49 @@
...
+        .do(({ errorMessage }) => Alert.alert('Error', `error occurred: ${errorMessage}`))
+        .ignoreElements();
 
 export const loadDepartmentsEpic$ = (action$: ActionsObservable<LoadDepartments>) =>
     action$.ofType('LOAD-DEPARTMENTS')
         .switchMap(x => ajaxGetJSON(`${url}/departments`))
         .map(x => deserializeArray(x as any, Department))
-        .map(x => loadDepartmentsFinished(x));
+        .map(x => loadDepartmentsFinished(x))
+        .catch(e => Observable.of({ type: 'ERROR-LOAD-FAILED', errorMessage: e.message || 'something wrong' }));
```
Comment (santee) [inline mobile/src/reducers/organization/organization.epics.ts]: you have a function that creates these instances, why do you create them manually? -> `errorLoadFailed(e.message)`

File: mobile/src/reducers/organization/organization.action.ts [hunk 8]
```diff
@@ -37,6 +37,13 @@ export interface LoadEmployeesForDepartment extends Action {
 
 export const loadEmployeesForDepartment = (departmentId: string): LoadEmployeesForDepartment => ({ type: 'LOAD_EMPLOYEES_FOR_DEPARTMENT', departmentId });
 
+export interface ErrorLoadFailed extends Action {
+    type: 'ERROR-LOAD-FAILED';
+    errorMessage: string;
+}
+export const errorloadFailed = (errorMessage: string): ErrorLoadFailed =>  ({ type: 'ERROR-LOAD-FAILED', errorMessage });
```
Comment (santee) [inline mobile/src/reducers/organization/organization.action.ts]: -> LoadFailedError

File: mobile/src/reducers/organization/organization.epics.ts [hunk 9]
```diff
@@ -1,40 +1,49 @@
...
 import { deserializeArray, deserialize } from 'santee-dcts/src/deserializer';
 import { Department } from './department.model';
 import { ajaxGetJSON } from 'rxjs/observable/dom/AjaxObservable';
 import { AppState } from '../app.reducer';
 import { Employee } from './employee.model';
+import { Observable } from 'rxjs/Observable';
+import { Alert } from 'react-native';
 
 const url = 'http://localhost:5000/api'; //TODO: fix hardcode
+export const errorEpic$ = (action$: ActionsObservable<ErrorLoadFailed>) =>
```
Comment (santee) [inline mobile/src/reducers/organization/organization.epics.ts]: let's move errors related code to a separate folder, not in organization

File: mobile/src/reducers/error/error.action.ts [hunk 10]
```diff
@@ -0,0 +1,6 @@
+import { Action } from 'redux';
```
Comment (serjKim) [inline mobile/src/reducers/error/error.action.ts]: Rename file to `error-load-failed.action.ts`

File: mobile/src/reducers/organization/organization.action.ts [hunk 11]
```diff
@@ -37,12 +37,6 @@ export interface LoadEmployeesForDepartment extends Action {
 
 export const loadEmployeesForDepartment = (departmentId: string): LoadEmployeesForDepartment => ({ type: 'LOAD_EMPLOYEES_FOR_DEPARTMENT', departmentId });
 
-export interface ErrorLoadFailed extends Action {
-    type: 'ERROR-LOAD-FAILED';
-    errorMessage: string;
-}
-export const errorloadFailed = (errorMessage: string): ErrorLoadFailed =>  ({ type: 'ERROR-LOAD-FAILED', errorMessage });
```
Comment (serjKim) [inline mobile/src/reducers/organization/organization.action.ts]: It should be used. And move it to the `error-load-failed.action.ts` and use for:
```
...
.catch(e => Observable.of(errorLoadFailed(e.message)))
```

File: mobile/src/reducers/organization/organization.epics.ts [hunk 12]
```diff
@@ -1,5 +1,6 @@
 import { ActionsObservable } from 'redux-observable';
```
Comment (serjKim) [inline mobile/src/reducers/organization/organization.epics.ts]: Move `$errorEpic$` to the separate file `error-load-failed.epic.ts` and rename to `errorLoadFailedEpic$`

File: mobile/src/reducers/error/error.action.ts (hunk 10 above)
Comment (santee) [reply mobile/src/reducers/error/error.action.ts]: there is no need, i think, do you want to have a separate file for each potential error action? I'd just rename it to errors/errors

Comment (serjKim) [reply mobile/src/reducers/error/error.action.ts]: ok

File: mobile/src/reducers/organization/organization.epics.ts [hunk 13]
```diff
@@ -1,26 +1,21 @@
...
-        .do(({ errorMessage }) => Alert.alert('Error', `error occurred: ${errorMessage}`))
-        .ignoreElements();
 
 export const loadDepartmentsEpic$ = (action$: ActionsObservable<LoadDepartments>) =>
     action$.ofType('LOAD-DEPARTMENTS')
         .switchMap(x => ajaxGetJSON(`${url}/departments`))
         .map(x => deserializeArray(x as any, Department))
         .map(x => loadDepartmentsFinished(x))
-        .catch(e => Observable.of({ type: 'ERROR-LOAD-FAILED', errorMessage: e.message || 'something wrong' }));
+        .catch(e => Observable.of(errorLoadFailed(e.message || 'something wrong')));
```
Comment (serjKim) [inline mobile/src/reducers/organization/organization.epics.ts]: Put 'something wrong' into `errorLoadFailed`

```const errorLoadFailed = (errorMessage: string): ErrorLoadFailed => ({ type: 'ERROR-LOAD-FAILED', errorMessage: errorMessage ||  'something wrong'})```

File: mobile/src/reducers/organization/organization.reducer.ts [hunk 14]
```diff
@@ -15,7 +16,8 @@ export const organizationEpics = combineEpics(
     loadDepartmentsEpic$ as any,
     loadChiefsEpic$ as any,
     loadDepartmentsFinishedEpic$ as any,
-    loadEmployeesForDepartmentEpic$ as any);
+    loadEmployeesForDepartmentEpic$ as any,
+    errorLoadFailedEpic$ as any);
```
Comment (serjKim) [inline mobile/src/reducers/organization/organization.reducer.ts]: Also, move epic to the `errors\errors.reducer.ts` and add to the root epic:
```
export const errorsEpics = combineEpics(errorLoadFailedEpic$)
```
app.reducer.ts:
```
const rootEpic = combineEpics( helpdeskEpics as any, organizationEpics as any, errorsEpics);
```

**Pull request merged**